src/app/products/manage-tax-components/create-tax-component/create-tax-component.component.ts

```diff
@@ -0,0 +1,148 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { ProductsService } from '../../products.service';
+
+/**
+ * Create Tax Component component.
+ */
+@Component({
+  selector: 'mifosx-create-tax-component',
+  templateUrl: './create-tax-component.component.html',
+  styleUrls: ['./create-tax-component.component.scss']
+})
+export class CreateTaxComponentComponent implements OnInit {
+
+  /** Minimum start date allowed. */
+  minDate = new Date();
+  /** Maximum start date allowed. */
+  maxDate = new Date(2025, 6, 22);
+  /** Tax Component form. */
+  taxComponentForm: FormGroup;
+  /** Tax Component template data. */
+  taxComponentTemplateData: any;
+  /** Credit Account Type data. */
+  creditAccountTypeData: any;
+  /** Credit Account data. */
+  creditAccountData: any;
+  /** Debit Account Type data. */
+  debitAccountTypeData: any;
+  /** Debit Account data. */
+  debitAccountData: any;
+
+  /**
+   * Retrieves the tax Component template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {ProductsService} productsService Products Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   * @param {DatePipe} datePipe Date Pipe to format date.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private productsService: ProductsService,
+              private route: ActivatedRoute,
+              private router: Router,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe(( data: { taxComponentTemplate: any }) => {
+      this.taxComponentTemplateData = data.taxComponentTemplate;
+    });
+  }
+
+  /**
+   * Creates the tax Component form
+   */
+  ngOnInit() {
+    this.createTaxComponentForm();
+    this.setCreditAccountsForm();
+    this.setDebitAccountsForm();
```

Comment:
Add a single function to set conditional controls. Add/Remove control conditional instead of update for this case.

For example, https://github.com/openMF/web-app/blob/master/src/app/users/create-user/create-user.component.ts